fix: RWD & Keyboard Access. for "Home" & "Our Projects" Carousels

jschhie opened the pull request:
# Reference
- [Issue #85](https://github.com/Crustaly/audemywebsite/issues/85): Keyboard access. for `<Carousel/>`

--- 

# Summary
## Key Changes
- Fix RWD for `<Testimonials/>` carousel on "Home" page
  - **Before:**
    - Narrow text/cards & unequal images during resizing
- Fix RWD for `<Carousel/>` on "Our Projects" page
- Fix keyboard accessibility for "Our Projects" `<Carousel/>`
  - **Before:**
    - Buttons = not tabbable; did not respond to <kbd>Enter</kbd>
  - **Approach:**
    - Replace static values with reactive constants
    - Add semantic, functional `<button @click="...">` 
    - Add bindings to new `goToPrevSlide()` and `goToNextSlide()` functions
    - Add image mappings (`src` & `alt`) for dynamic `<Carousel/>` SR output

## Supporting updates
<details>
  <summary>View</summary>

- **Circular Behavior Fix: `<Carousel/>`**
  - **Before:** <kbd>Previous</kbd> button did nothing on the first image
  - **After:** Fast-forwards to last image; aligns with <kbd>Next</kbd> button's behavior

- **Fix Screen Reader Access in "Home"**
  - Hide icons and images from SR output
  - Hide redundant `<PressListCard/>` links from the tab order 

- **Minor UI updates, refactor, & DRY classes**
  - Format text and emails for UI readability (via `v-html`)
  - Add scrolling to `<Testimonials/>`

</details>

  
---

# Previews

## Home: `<Testimonials/>` 

<details>
  <summary>View screenshots (RWD)</summary>

### Before & After: Mobile - Medium RWD 

| Before | After | 
|:---: |:---: |
| <img src="https://github.com/user-attachments/assets/eef02604-c9e6-4a7d-b4fa-6e5483a98a25" width="300" /> | <img width="300" alt="mobile small" src="https://github.com/user-attachments/assets/47f5464a-be43-4e3e-ae80-127f8774e28e" /> | 

</details>

<details>
<summary>View live demo (RWD + Enabled Scrolling)</summary>

### Live Demo: RWD + Scrolling
| Mobile - Medium |
|:---: |
| <video src="https://github.com/user-attachments/assets/63e33f9b-8f9b-43a4-bcb4-14aafaa93d02" width="250" controls /> |

</details>

---

## Our Projects: `<Carousel/>`

<details>
  <summary>View demos (RWD & UI Updates)</summary>

### RWD & UI Updates

| Before | After |  
|:---: | :---: |
| <img width="300" alt="Screenshot 2025-08-31 at 5 18 11 PM" src="https://github.com/user-attachments/assets/6fc260b6-fbed-4e54-90b5-376840d6a68b" /> | <img src="https://github.com/user-attachments/assets/4d3d3508-95e7-47e9-930e-0f397633f285" width="300" /> | 

| Medium | Large - 2XL | 
|:---: |:---: | 
| <img src="https://github.com/user-attachments/assets/4744a2a9-c2a1-49b6-8132-991a9c9c0d3a" width="500" /> | <img src="https://github.com/user-attachments/assets/50f72f8a-b371-4a94-bbcc-4bd04cd81fd4" width="500" /> |

</details>


<details>
<summary>View live demo (Keyboard Access)</summary>

### Live Demo
| Mobile - Medium |
|:---: |
| <video src="https://github.com/user-attachments/assets/ee6cbeb2-7e06-4b55-a133-f7add2e3fb9d" width="500" controls /> |

</details>

---

# Manual Tests

<details>
  <summary>View</summary>

- **RWD:** 
  - Test screen widths from `375px` (**mobile**) to `1680px` (**2xl**)
- **Lighthouse Audit:** 
  - No accessibility issues for `<Testimonials/>` & `<Carousel/>`
  - Carousel buttons are accessible via <kbd>Tab</kbd> and <kbd>Enter</kbd>
- **Home:**
  - Redundant `<PressListCard/>` images are skipped in the tab order
  - Focus correctly goes to text-based links 
- **Our Projects:**
  - **Circular Navigation:** <kbd>Previous</kbd> button from first image skips to last
  
</details>

jschhie committed:
refactor: DRY UI, format <Testimonials/>, & accessibility fixes

- Format testimonial text/emails for UI readability (via v-html)
- Add DRY style classes for page content & btns
- Minor SR accessibility: Hide imgs; add aria-labels to visual arrow btns

jschhie committed:
fix: RWD & add <Testimonials/> carousel scrolling

- Fix RWD for cards: column (mobile-md) & row (lg+)
- Add RWD for arrow btns: up/down (mobile-md) & left/right (lg+)
- Add vertical (mobile-md) & horizontal (lg+) scrolling to carousel
- Add 2xl RWD for <Testimonial/> buttons, text, & images
- Style: hover effects & RWD transitions

jschhie committed:
accessibility: Hide redundant linked <PressListCard/> images & remove them from tab order

- Hide images & remove their repeated links from SR output

jschhie committed:
refactor: RWD & DRY UI in <OurProjects/> carousel

- Fix RWD to avoid icon/image overlap in <ProjectsInAction/>
- Improve RWD spacing/resizing in <OurProjects/> & <Carousel/>
- Add DRY classes & update UI in <ProjectsInAction/> & <Carousel/>
- Accessibility: Hide decorative images & <Carousel/> from SR output

jschhie committed:
fix: Keyboard access for Our Projects <Carousel/>

- Fix UI logic for <Carousel/> buttons to be accessible via tab & Enter
- Replace inaccessible <img/> & complex event listeners:
  - Replace static values with reactive currentSlide, carousel, & maxLength
  - Add semantic <button @click=...> & goToNext/PrevSlide() functions
- Add image mappings (src & alt text) for dynamic <Carousel/> SR output
- Update <Carousel/> for better RWD, DRY code, & UI consistency

jschhie committed:
fix: Enable <Carousel/> previous button circularity

- Before: Previous button on first image did nothing
- After: Circular behavior; fast forwards to last image
- to align with Next button on last image (back to first)

jschhie commented:
Hi @master-of-none, 

Could you please review this PR whenever you get a chance? 

It includes RWD & keyboard accessibility fixes for the "Home" & "Our Projects" carousels.  

Thank you!

master-of-none commented:
- Lighthouse test works Good. 
- Good UI/UX update 

@jschhie Is it possible to make the lines, not overlap on text. Like lines hide behind the text, kind of a layering mechanism. 


<img width="354" height="405" alt="image" src="https://github.com/user-attachments/assets/31392eea-0ea6-4257-b8b9-cb55eabc39b0" />

jschhie committed:
fix: <Testimonials/> & <GamesShowcase/> legibility

- Add white BG to text in both carousels
- Center text in <Testimonials/> for UI balance
- Minor: + hover effect to <GamesShowcase/> to align UI w/ <Testimonials/>

jschhie commented:
Thanks for the feedback! @master-of-none 

Just rebased with `preprod` & updated the PR for better legibility in carousels.

It's ready for review again whenever you can 👍

| `<Testimonials/>`  | `<GamesShowcase/>` |
| :--: | :--: | 
| <img width="500" alt="Screenshot 2025-08-31 at 8 53 03 PM" src="https://github.com/user-attachments/assets/1094b776-5d99-4e29-a04b-44ae94c72d90" /> | <img width="500" alt="Screenshot 2025-08-31 at 8 35 37 PM" src="https://github.com/user-attachments/assets/39a5e805-5d6b-4e3e-94b1-a08fcaa7c346" /> |

master-of-none reviewed (approved):
Changes Approved for merge
